src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -162,6 +162,13 @@ object Parameters {
   val PARAM_INTERNAL_CHECK_TABLE_EXISTENCE_IN_CURRENT_SCHEMA_ONLY: String = knownParam(
     "internal_check_table_existence_in_current_schema_only"
   )
+  // Internal option to skip write operation when writing a DataFrame
+  // without any partitions. By default, it it is 'false'.
+  // This option may be removed without any notice in any time.
+  // Introduced since Spark Connector 2.8.5
+  val PARAM_INTERNAL_SKIP_WRITE_WHEN_WRITING_EMPTY_DATAFRAME: String = knownParam(
+    "internal_skip_write_when_writing_empty_dataframe"
+  )
```

Comment:
why we need this parameter? this PR fixes a bug right? it is not a new feature

Reply:
I agree this is bug fix. Even if it is bug fix, it is a behaviour change.
Our current strategy is to introduce an internal option to disable the behavior change.
If any customer depends on the OLD behaviour, they can use this internal option.

Reply:
is it a behavior change? 
Change from not support empty DataFrame to support empty DataFrame is not a behavior change.
It is a bug fix or kinds of new feature.

Reply:
@sfc-gh-sshankar @sfc-gh-ema what do you think?

Reply:
I think if we use parameters like this, we need to have a long-term plan to remove these, or else we will end up with too many parameters.